File: app/src/main/java/com/botijasoftware/utils/interpolator/CosInterpolator.java
package com.botijasoftware.utils.interpolator;

public class CosInterpolator implements Interpolator {
	//public abstract Interpolator() {}
	public float interpolate(float v0, float v1, float t) {
		float a = (1-(float)Math.cos(t*(float)Math.PI))/2.0f;
		return v0 + a*(v1 - v0);
	}
}

File: app/src/main/java/com/botijasoftware/utils/interpolator/CosInterpolator.kt
package com.botijasoftware.utils.interpolator

class CosInterpolator : Interpolator {
    //public abstract Interpolator() {}
    override fun interpolate(v0: Float, v1: Float, t: Float): Float {
        val a = (1 - Math.cos((t * Math.PI.toFloat()).toDouble()).toFloat()) / 2.0f
        return v0 + a * (v1 - v0)
    }
}
